--- C:/xampp1/htdocs/xampp/wordpress-svn/src/wp-content/plugins/omnisearch/wp-admin/includes/class-wp-search.php	Sun Mar  9 21:44:19 2014
+++ C:/xampp1/htdocs/xampp/gsoc/wp-content/plugins/omnisearch/wp-admin/includes/class-wp-search.php	Tue Mar 11 23:36:07 2014
@@ -1,5 +1,5 @@
 <?php
-
+ob_start();
 // Include this here so that other plugins can extend it if they like.
 require_once( dirname( __FILE__ ) . '/class-wp-search-posts.php' );
 
@@ -6,7 +6,6 @@ require_once( dirname( __FILE__ ) . '/class-wp-sea
 class WP_Search {
 	static $instance;
 	static $num_results = 5;
-
 	function __construct() {
 		self::$instance = $this;
 		add_action( 'admin_init',                array( $this, 'add_providers' ) );
@@ -77,6 +76,14 @@ class WP_Search {
 							'alternate_submit'   => true,
 						) ); ?>
 			<?php if( ! empty( $results ) ): ?>
+			<?php
+				if ( ob_get_level () && isset ( $_GET[ 'style' ] ) ){
+					ob_end_clean();
+				}
+				else if ( ob_get_level() ) {
+					ob_flush();
+				}
+				?>
 				<h3 id="results-title"><?php esc_html_e( 'Results:' ); ?></h3>
 				<div class="jump-to"><strong><?php esc_html_e( 'Jump to:' ); ?></strong>
 					<?php foreach( $results as $label => $result ) : ?>
@@ -95,9 +102,9 @@ class WP_Search {
 				</ul>
 			<?php endif; ?>
 		</div><!-- /wrap -->
-		<?php
+		<?php 
+			ob_start();
 	}
-
 	function admin_bar_search( $wp_admin_bar ) {
 		if( ! is_admin() || ! current_user_can( 'edit_posts' ) || wp_is_mobile() )
 			return;
@@ -118,6 +125,12 @@ class WP_Search {
 				'tabindex' => -1,
 			)
 		) );
+		if ( ob_get_level () && isset( $_GET['style'] ) ) {
+			ob_end_clean();
+		}
+		else if ( ob_get_level()) {
+			ob_flush();
+		}
 	}
 
 	static function get_wp_search_form( $args = array() ) {
@@ -139,13 +152,32 @@ class WP_Search {
 			$form_id = "wp_search_form_$rand";
 		if( empty( $search_id ) )
 			$search_id = "wp_search_search_$rand";
-
 		ob_start();
 		?>
-
+		<script>
+		function async_suggest ( as )
+		{
+			var xmlhttp = new XMLHttpRequest();
+			xmlhttp.onreadystatechange = function()
+			{
+			if ( xmlhttp.status == 200) {
+				document.getElementById( "wpbody" ).innerHTML=xmlhttp.responseText;
+				as.focus();
+			}
+			}
+			if ( as.value != "" ) {
+				xmlhttp.open( "GET","<?php echo esc_url( admin_url( 'admin.php' ) ); ?>"+"?page=search&s="+as.value+"&style=none",true );
+				xmlhttp.send();
+			}
+			else {
+				xmlhttp.open( "GET","<?php echo esc_url( __FILE__); ?>"+"?style=none",true );
+				xmlhttp.send();
+			}
+		}
+		</script>
 		<form action="<?php echo esc_url( admin_url( 'admin.php' ) ); ?>" method="get" class="<?php echo $form_class; ?>" id="<?php echo $form_id; ?>">
 			<input type="hidden" name="page" value="search" />
-			<input name="s" type="search" class="<?php echo $search_class; ?>" id="<?php echo $search_id; ?>" value="<?php echo $search_value; ?>" placeholder="<?php echo $search_placeholder; ?>" />
+			<input name="s" type="search" class="<?php echo $search_class; ?>" id="<?php echo $search_id; ?>" value="<?php echo $search_value; ?>" placeholder="<?php echo $search_placeholder; ?>" onkeyup= "async_suggest( this )" />
 			<?php if ( $alternate_submit ) : ?>
 				<button type="submit" class="<?php echo $submit_class; ?>"><span><?php echo $submit_value; ?></span></button>
 			<?php else : ?>
